### PR TITLE
E2e test script

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- an initial version of a script that can be used to run end to end tests

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/2271
**Related PR(s)**
#2619

### CODE REVIEW

**Comment by k15r** [comment]
@sjanota imho the difference to other test is: 
normally you install a release and it stays in the cluster. The e2e-test currently use helm just out of pure convenience. The do not stay in the cluster as a component of kyma. Their release will be installed, the tests will be executed, the release will be removed. In the future helm will probably get replaced by a different mechanism.

**Comment by sjanota** [comment]
* Our acceptance tests rarely test single component in isolation. Some of them are already e2e tests rather then integration tests. They are bound to a component by test entry point (not documented anywhere, it's my gut feeling). For example kubeless-integration tests test lambda creation. It involves at least kubeless, API controller and istio and verifies complete user scenario.
* They stay in cluster not because they are bound to come component, but because you can easily rerun them if needed without the need to have Kyma sources. Binding is a way of achieving that, not the goal itself. 
* New mechanism to run our tests is currently under development (#2639) and all acceptance tests (no matter integration or e2e) should use it.

**Comment by sjanota** [comment]
FYI @PK85

**Comment by aszecowka** [comment]
why do we have 2 files with tests: testing.sh and e2e-testing.sh?

**Comment by venturasr** [comment]
@aszecowka `e2e-testing.sh` is the script we are using for executing only end to end test. As you suggested in the PR, https://github.com/kyma-project/test-infra/pull/583 . we have created a `kyma-gke-end-to-end-test.sh` to provisioning cluster on for the execution of end to end test.

**Comment by joek** [comment]
> * Our acceptance tests rarely test single component in isolation. Some of them are already e2e tests rather then integration tests. They are bound to a component by test entry point (not documented anywhere, it's my gut feeling). For example kubeless-integration tests test lambda creation. It involves at least kubeless, API controller and istio and verifies complete user scenario.
> * They stay in cluster not because they are bound to come component, but because you can easily rerun them if needed without the need to have Kyma sources. Binding is a way of achieving that, not the goal itself.
> * New mechanism to run our tests is currently under development (#2639) and all acceptance tests (no matter integration or e2e) should use it.

Actually I would suggest, we are going with the solution how it is implemented right now. As soon as we have the new process in place we will migrate to the new standard. I don't know the technical implications but my experience with helm test is not the best. Especially if we have a look on the tests testing more then one component.

**Comment by sjanota** [comment]
@joek @k15r @venturasr I'd like to understand what's so different about those tests that we need to bring whole new infrastructure to run them. Can you point me to already implemented e2e test?

**Comment by venturasr** [comment]
@sjanota 
Current implementation of Backup E2E Test:

    https://github.com/kyma-project/test-infra/pull/584
    https://github.com/kyma-project/test-infra/pull/583
    https://github.com/kyma-project/kyma/pull/2646
    https://github.com/kyma-project/kyma/pull/2619

Test Modularisation plan:

    https://github.com/kyma-project/kyma/issues/2696

File: installation/scripts/e2e-testing.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+cleanupE2EErr=$?
+
+
+
+echo "----------------------------"
+echo "- E2E Testing Kyma..."
+echo "----------------------------"
+
+
+for testcase in ../../tests/end-to-end/*/deploy/chart/*
```
Comment (aszecowka) [inline installation/scripts/e2e-testing.sh]: could you point me what will be tested here? I cannot find this files on the master branch

Comment (venturasr) [reply installation/scripts/e2e-testing.sh]: Hi @aszecowka, the whole implementation of the e2e test is in this PR. 
https://github.com/kyma-project/kyma/pull/2619
However we are working on the modularization of the e2e test as it is explained in this issue, https://github.com/kyma-project/kyma/issues/2696

**Comment by sjanota** [comment]
@venturasr The test implemented in #2619 is simply an acceptance test, there's nothing special about it. It operates on it's own resources, have it's own namespace and does not collide with other subsystems. It could be as well run as simple helm test attached to ark chart. There is no need to implement new way of running tests for this scenario.

**Comment by venturasr** [comment]
@sjanota and @aszecowka thank you for your help guys

**Pull request merged**